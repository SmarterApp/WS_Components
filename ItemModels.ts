import * as ItemCard from "./ItemCard";
import * as $ from 'jquery';
import * as GradeLevels from './GradeLevels';
import * as API from './ApiModels';

export interface ScoreSearchParams {
    gradeLevels: GradeLevels.GradeLevels;
    subjects: string[];
    techType: string[];
}

export const ScoreSearchClient = (params: ScoreSearchParams) => API.get<ItemCard.ItemCardViewModel[]>("/ScoringGuide/Search", params) 
