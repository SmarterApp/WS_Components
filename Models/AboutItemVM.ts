import * as Rubric from '../PageTabs/Rubric'
import * as ItemCardViewModel from './ItemCardViewModel'
import * as API from '../Models/ApiModels'

export const ScoreSearchClient = (params: { bankKey: number, itemKey: number }) => API.get<AboutThisItem>("/api/aboutItem", params)

export interface AboutThisItem {
    rubrics: Rubric.Rubric[];
    itemCardViewModel: ItemCardViewModel.ItemCardViewModel;
    depthOfKnowledge: string;
    targetDescription: string;
    commonCoreStandardsDescription: string;
    educationalDifficulty: string;
    evidenceStatement: string;
}