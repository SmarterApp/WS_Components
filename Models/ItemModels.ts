import * as $ from 'jquery';
import { GradeLevels } from './GradeLevels';
import * as API from './ApiModels';
import * as ItemCardViewModel from './ItemCardViewModel';

export interface Subject {
    code: string;
    label: string;
}

export interface TechType extends Subject { }

export interface FilterOptions {
    subjects: Subject[];
    grades: GradeLevels[];
    techTypes: TechType[];
}

export interface ItemFilter extends FilterOptions { };