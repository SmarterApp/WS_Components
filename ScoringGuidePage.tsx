import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as ItemViewerFrame from './ItemViewerFrame';
import * as ItemModels from './ItemModels';
import * as ApiModels from './ApiModels';
import * as ItemCard from './ItemCard';
import * as GradeLevels from './GradeLevels';
import * as ItemCardViewer from './ItemCardViewer';
import * as AboutItem from './AboutItem';
import * as ItemTable from './ItemTable';
import * as ItemSearchDropdown from './ItemSearchDropdown';
import * as PageTabs from './PageTabs'
import { get } from "./ApiModels";


const SearchClient = (params: ItemSearchDropdown.SearchAPIParams) => get<ItemCard.ItemCardViewModel[]>("http://is-score.cass.oregonstate.edu/ScoringGuide/Search", params);
const ScoreGuideViewModelClient = () => get<ItemsSearchViewModel>("http://is-score.cass.oregonstate.edu/ScoringGuide/ScoringGuideViewModel");

export interface State {
    searchParams: ItemModels.ScoreSearchParams;
    itemSearchResult: ApiModels.Resource<ItemCard.ItemCardViewModel[]>;
    selectedItem: ApiModels.Resource<AboutItem.AboutThisItem>;
    selectedRow?: ItemCard.ItemCardViewModel; 
    sorts: ItemTable.HeaderSort[];
    scoringGuideViewModel: ApiModels.Resource<ItemsSearchViewModel>;
}

export interface ItemsSearchViewModel {
    interactionTypes: ItemSearchDropdown.InteractionType[];
    subjects: ItemSearchDropdown.Subject[];
}
 
export class ScoringGuidePage extends React.Component<{}, State> {
    private headerColumns = ItemTable.headerColumns;
    private dataTableRef: HTMLTableElement;

    constructor() {
        super();
        this.state = {
            searchParams: {
                gradeLevels: GradeLevels.GradeLevels.NA,
                subjects: [],
                techType: []
            },
            itemSearchResult: { kind: "loading" },
            selectedItem: { kind: "loading" },
            sorts: [],
            scoringGuideViewModel: {kind: "loading"}
        }

        this.loadScoringGuideViewModel();
    }

    loadScoringGuideViewModel(){
        ScoreGuideViewModelClient()
        .then(result => this.onSuccessLoadScoringGuideViewModel(result))
        .catch(err => this.onErrorLoadScoringGuideViewModel(err));

    }

    onSuccessLoadScoringGuideViewModel(result: ItemsSearchViewModel){
        this.setState({
            scoringGuideViewModel: {kind:"success", content: result}
        })
    }

    onErrorLoadScoringGuideViewModel(err: any){
        console.error(err);
    }

    //row is selected passed from item table
    onSelectItem = (item: ItemCard.ItemCardViewModel) => {
        this.setState({
            selectedRow: item
        });

        AboutItem.ScoreSearchClient({ bankKey: item.bankKey, itemKey: item.itemKey })
            .then((data) => this.onAboutItemSuccess(data))
            .catch((err) => this.onAboutItemError(err));
    };

    //on load success, after row is selected
    onAboutItemSuccess(item: AboutItem.AboutThisItem) {
        this.setState({
            selectedItem: { kind: "success", content: item }
        });
    }

    onAboutItemError(err: any) {
        console.error(err);
    }

    onSearchSuccess(result: ItemCard.ItemCardViewModel[]) {
        const searchParams = this.state.searchParams;
        const items = result;
        this.setState({
            itemSearchResult: { kind: "success", content: items },
        });
    }

    onSearchError(err: any) {
        console.log(err);
    }

    async callSearch() {
        const searchParams = this.state.searchParams;
        return ItemModels.ScoreSearchClient(searchParams)
            .then((data) => this.onSearchSuccess(data))
            .catch((err) => this.onSearchError(err));
    }

    isLoading() {
        return this.state.itemSearchResult.kind === "loading" || this.state.itemSearchResult.kind === "reloading";
    }

    renderAboutItemDetails() {
        const selectedResult = this.state.selectedItem;
        if (selectedResult.kind == "success" && selectedResult.content) {
            const itemCard = selectedResult.content.itemCardViewModel;
            return (
                <div>
                    <ItemCardViewer.ItemCardViewer aboutItem= {selectedResult.content} />
                </div>
            );
        } else {
            return (<div></div>);
        }

    }

    onClickHeader = (col: ItemTable.SortColumn) => {
        const newSorts = (this.state.sorts || []).slice();
        const headIdx = newSorts.findIndex(hs => hs.col.header === col.header);
        if (headIdx !== -1) {
            const newSort = Object.assign({}, newSorts[headIdx]);
            if (newSort.direction == ItemTable.SortDirection.Ascending) {
                newSort.direction = ItemTable.SortDirection.Descending;
            }
            else if (newSort.direction == ItemTable.SortDirection.Descending) {
                newSort.direction = ItemTable.SortDirection.NoSort;
            }
            else {
                newSort.direction = ItemTable.SortDirection.Ascending;
            }

            newSorts[headIdx] = newSort;
        } else {
            const newSort: ItemTable.HeaderSort = {
                col: col,
                direction: ItemTable.SortDirection.Ascending,
                resetSortCount:0
            };
            newSorts.push(newSort);
        }
        this.setState({ sorts: newSorts });
    }

    invokeMultiSort(lhs: ItemCard.ItemCardViewModel, rhs: ItemCard.ItemCardViewModel): number {
        const sorts = this.state.sorts || [];
        for (const sort of sorts) {
            const diff = sort.col.compare(lhs, rhs) * sort.direction;
            if (diff !== 0) {
                return diff;
            }
        }
        return 0;
    }

    //Post sorted table data.
    getTableData(data: ItemCard.ItemCardViewModel[]): ItemCard.ItemCardViewModel[] {
        const sortedData = this.state.sorts && this.state.sorts.length !== 0
            ? data.sort((lhs, rhs) => this.invokeMultiSort(lhs, rhs))
            : data;

        return sortedData;
    }

    onSearch(results: ItemCard.ItemCardViewModel[]) {
        this.setState({ itemSearchResult: { kind: "success", content: results } });
    }

    onError(err: any) {
        this.setState({ itemSearchResult: { kind: "failure" } });
        console.error(err);
    }

    selectSingleResult() {
        const searchResults = this.state.itemSearchResult;
        if (searchResults.kind === "success" && searchResults.content!.length === 1) {
            const searchResult = searchResults.content![0];
            ItemCard.itemPageLink(searchResult.bankKey, searchResult.itemKey);
        }
    }

    beginSearch(params: ItemSearchDropdown.SearchAPIParams) {
        const searchResults = this.state.itemSearchResult;
        if (searchResults.kind === "success") {
            this.setState({
                itemSearchResult: {
                    kind: "reloading",
                    content: searchResults.content
                }
            });
        } else if (searchResults.kind === "failure") {
            this.setState({
                itemSearchResult: { kind: "loading" }
            });
        }

        SearchClient(params)
        .then(data => this.onSearchSuccess(data))
        .catch(err => this.onSearchError(err));
    }

    renderSearchControls(isLoading: boolean) {
        const vmState = this.state.scoringGuideViewModel;
        if((vmState.kind == "success" || vmState.kind == "reloading") && vmState.content != undefined){
            return (
                <div className="search-controls">
                    <a>Print Items</a>
                    <ItemSearchDropdown.ItemSearchDropdown
                        interactionTypes={vmState.content.interactionTypes}
                        subjects={vmState.content.subjects}
                        onChange={(params) => this.beginSearch(params)}
                        selectSingleResult={() => this.selectSingleResult()}
                        isLoading={isLoading} />
                </div>
            );
        }else{
            return null;
        }
    
    }

    renderSearch() {
        const searchResults = this.state.itemSearchResult;

        let resultElement: JSX.Element[] | JSX.Element | undefined;
        if (searchResults.kind === "success" || searchResults.kind === "reloading") {
            if (searchResults.content == null || searchResults.content.length === 0) {
                resultElement = <span className="placeholder-text" role="alert">No results found for the given search terms.</span>
            }
            else {
                resultElement =
                    <div className="search-results">

                        <ItemTable.HeaderTable
                            sorts={this.state.sorts}
                            onHeaderClick={this.onClickHeader}
                            columns={this.headerColumns} />
                        <ItemTable.DataTable
                            mapRows={this.getTableData(searchResults.content)}
                            rowOnClick={this.onSelectItem}
                            sort={this.state.sorts}
                            tableRef={ref => this.dataTableRef = ref}
                            columns={this.headerColumns}
                            selectedRow={this.state.selectedRow} />
                    </div>
            }
        } else if (searchResults.kind === "failure") {
            resultElement = <div className="placeholder-text" role="alert">An error occurred. Please try again later.</div>;
        }

        return resultElement;
    }

    render() {

        const isLoading = this.isLoading();
        return (
            <div className="search-page">
                <div className="search-container">
                    {this.renderSearchControls(isLoading)}
                    {this.renderSearch()}
                </div>
                {this.renderAboutItemDetails()}
            </div>
        );
    }
}

export function initScoreGuidePage(){
    ReactDOM.render(<ScoringGuidePage />, document.getElementById("react-container"));
}

