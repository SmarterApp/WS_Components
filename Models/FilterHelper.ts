import { ItemCardViewModel } from "../Models/ItemCardViewModel";
import { ItemFilter, FilterOptions } from "../Models/ItemModels";
import * as GradeLevels from '../Models/GradeLevels';
import { parseQueryString } from "../Models/ApiModels";
import { AdvancedFilterCategory, AdvancedFilters, OptionType, AdvancedFilterOption } from "../filter/AdvancedFilterModel";

export class FilterHelper {

    //TODO: Get this from the server
    static getFilterOptions(): AdvancedFilters  {
        const subjectsFilterOptions: AdvancedFilterOption[] = [{
            label: "Mathematics",
            key: "MATH",
            isSelected: false,
            type: OptionType.button
        },
        {
            label: "English",
            key: "ELA",
            isSelected: false,
            type: OptionType.button
        }];

        const gradesFilterOptions: AdvancedFilterOption[] = [{
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Elementary),
            key: String(GradeLevels.GradeLevels.Elementary),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade3),
            key: String(GradeLevels.GradeLevels.Grade3),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade4),
            key: String(GradeLevels.GradeLevels.Grade4),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade5),
            key: String(GradeLevels.GradeLevels.Grade5),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade6),
            key: String(GradeLevels.GradeLevels.Grade6),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade7),
            key: String(GradeLevels.GradeLevels.Grade7),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.Grade8),
            key: String(GradeLevels.GradeLevels.Grade8),
            isSelected: false,
            type: OptionType.button
        }, {
            label: GradeLevels.caseToString(GradeLevels.GradeLevels.High),
            key: String(GradeLevels.GradeLevels.High),
            isSelected: false,
            type: OptionType.button
        }];

        const techTypesFilterOptions: AdvancedFilterOption[] = [{
            label: "CAT",
            key: "CAT",
            isSelected: false,
            type: OptionType.button
        }, {
            label: "Performance Items",
            key: "PT",
            isSelected: false,
            type: OptionType.button
        }];

        const subjects: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "Subjects",
            helpText: "Subjects HelpText here.",
            filterOptions: [...subjectsFilterOptions],
            displayAllButton: true
        }

        const grades: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "Grade",
            helpText: "Grade HelpText here.",
            filterOptions: [...gradesFilterOptions],
            displayAllButton: true
        }

        const techTypes: AdvancedFilterCategory = {
            disabled: false,
            isMultiSelect: false,
            label: "TechType",
            helpText: "TechType HelpText here.",
            filterOptions: [...techTypesFilterOptions],
            displayAllButton: false
        }

        const value: AdvancedFilters = {subjects, grades, techTypes};
        
        return value;
    }

    static filter(itemCards: ItemCardViewModel[], filter: AdvancedFilterCategory[]): ItemCardViewModel[] {
        const grades = filter.find(afc => afc.label === "Grade");
        const subjects = filter.find(afc => afc.label === "Subjects");
        const techTypes = filter.find(afc => afc.label === "TechType");

        if (grades && grades.filterOptions) {
            let selectedGrades = GradeLevels.GradeLevels.NA;

            grades.filterOptions.forEach(gradeFilter => {
                if (gradeFilter.isSelected) {
                    selectedGrades = selectedGrades | Number(gradeFilter.key);
                }
            });

            if (selectedGrades === GradeLevels.GradeLevels.NA) {
                selectedGrades = GradeLevels.GradeLevels.All;
            }

            itemCards = itemCards.filter(g => GradeLevels.contains(selectedGrades, g.grade));
        }

        if (subjects && subjects.filterOptions) {
            let subjectCode:string|undefined;

            subjects.filterOptions.forEach(s => {
                if(s.isSelected){
                    subjectCode = s.key;
                }
            });
            if(subjectCode !== undefined){
                itemCards = itemCards.filter(i => subjectCode === i.subjectCode);
            }
        }
        //TODO: What is CAT technology? Filter? Ignore?
        if (techTypes && techTypes.filterOptions) {
            if (techTypes.filterOptions.find(t => t.key === "PT" && t.isSelected)) {
                itemCards = itemCards.filter(i => i.isPerformanceItem);
            } else if (techTypes.filterOptions.find(t => t.key === "CAT" && t.isSelected)) {
                itemCards = itemCards.filter(i => !i.isPerformanceItem);
            }
        }

        return itemCards;
    }

    static updateUrl(filter: AdvancedFilterCategory[]) {
        let pairs: string[] = [];

        const grades = filter.find(afc => afc.label === "Grade");
        const subjects = filter.find(afc => afc.label === "Subjects");
        const techTypes = filter.find(afc => afc.label === "TechType");

        if (grades && !grades.disabled) {
            const gradeString: string[] = [];

            grades.filterOptions.forEach(g => {
                if (g.isSelected) {
                    gradeString.push(g.key.toString());
                }
            });

            if (gradeString.length !== 0) {
                gradeString.join(',');
                pairs.push("gradeLevels=" + gradeString);
            }
        }
        if (subjects && !subjects.disabled) {
            const subjString: string[] = [];

            subjects.filterOptions.forEach(s => {
                if (s.isSelected) {
                    subjString.push(s.key);
                }
            });

            if (subjString.length !== 0) {
                subjString.join(',');
                pairs.push("subjects=" + subjString);
            }
        }
        if (techTypes && !techTypes.disabled) {
            const techTypesString: string[] = [];

            techTypes.filterOptions.forEach(t => {
                if (t.isSelected) {
                    techTypesString.push(t.key);
                }
            });
            if (techTypesString.length !== 0) {
                techTypesString.join(',');
                pairs.push("techTypes=" + techTypesString);
            }
        }

        let query: string;
        if (pairs.length === 0) {
            query = "/";
        } else {
            query = "?" + pairs.join("&");
        }
        history.replaceState(null, "", query);
    }

    static readUrl(filterOptions: AdvancedFilters) {
        const queryObject = parseQueryString(window.location.href);
        const subjects = queryObject["subjects"]
            ? queryObject["subjects"]!.map(
                subjCode => {
                    let subject = filterOptions.subjects.filterOptions.find(s => s.key === subjCode);
                    if (subject) {
                        return {
                            code: subject.key,
                            label: subject.label
                        }
                    }
                }) : [];

        const grades = queryObject["grades"]
            ? queryObject["grades"]!
                .map(gradeCode => Number(gradeCode) as GradeLevels.GradeLevels)
            : [];

        const techTypes = queryObject["techTypes"]
            ? queryObject["techTypes"]!.map(typeCode => {
                let techType = filterOptions.techTypes.filterOptions.find(t => t.key === typeCode);
                if (techType) {
                    return {
                        code: techType.key,
                        label: techType.label
                    }
                }
            }) : [];

        return {
            subjects: subjects,
            grades: grades,
            techTypes: techTypes
        } as ItemFilter;
    }
}