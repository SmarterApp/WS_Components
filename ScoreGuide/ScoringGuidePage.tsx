import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as ItemModels from '../Models/ItemModels';
import * as ApiModels from '../Models/ApiModels';
import * as GradeLevels from '../Models/GradeLevels';
import * as ItemCardViewer from '../AboutItem/ItemCardViewer';
import * as AboutItemVM from '../Models/AboutItemVM';
import * as ItemSearchContainer from './ItemSearchContainer';
import * as ItemCardViewModel from '../Models/ItemCardViewModel'
import { get } from "../Models/ApiModels";
import { FilterHelper } from "../Models/FilterHelper";

const SearchClient = () => get<ItemCardViewModel.ItemCardViewModel[]>("api/search");

export interface Props {
    scoreGuideViewModelClient: () => Promise<ItemsSearchViewModel>;
}

export interface State {
    item: ApiModels.Resource<AboutItemVM.AboutThisItem>;
    scoringGuideViewModel: ApiModels.Resource<ItemsSearchViewModel>;
    filterOptions: ItemModels.FilterOptions;
}

export interface ItemsSearchViewModel {
    subjects: ItemModels.Subject[];
}

export class ScoringGuidePage extends React.Component<Props, State> {
    constructor(props: Props) {
        super(props);
        this.state = {
            scoringGuideViewModel: { kind: "loading" },
            filterOptions: FilterHelper.getFilterOptions(),
            item: { kind: "none" }
        }
        this.loadScoringGuideViewModel();
    }


    getAboutItem = (item: { itemKey: number; bankKey: number }) => {
        AboutItemVM.ScoreSearchClient(item)
            .then((data) => this.onAboutItemSuccess(data))
            .catch((err) => this.onAboutItemError(err));
    }

    onAboutItemSuccess = (data: AboutItemVM.AboutThisItem) => {
        this.setState({
            item: { kind: "success", content: data }
        });
    }

    onAboutItemError = (err: any) => {
        console.error(err);
        this.setState({
            item: { kind: "failure" }
        })
    }

    loadScoringGuideViewModel = () => {
        this.props.scoreGuideViewModelClient()
            .then(result => this.onSuccessLoadScoringGuideViewModel(result))
            .catch(err => this.onErrorLoadScoringGuideViewModel(err));

    }

    onSuccessLoadScoringGuideViewModel = (result: ItemsSearchViewModel) => {
        this.setState({
            scoringGuideViewModel: { kind: "success", content: result },
            filterOptions: {
                ...this.state.filterOptions,
                subjects: result.subjects
            }
        });
    }

    onErrorLoadScoringGuideViewModel = (err: any) => {
        console.error(err);
        this.setState({
            scoringGuideViewModel: { kind: "failure" }
        });
    }

    onRowSelection = (item: { bankKey: number, itemKey: number }, reset: boolean) => {
        if (reset === false) {
            this.getAboutItem(item);
        }else {
            let state: ApiModels.Resource<AboutItemVM.AboutThisItem> = { kind: "none" };
            this.setState({ item: state })
        }
    }

    renderTabsContainer = () => {
        if (this.state.item.kind == "success" || this.state.item.kind == "reloading") {
            const newItem = this.state.item.content;
            return (
                <div>
                    <ItemCardViewer.ItemCardViewer
                        item={newItem}
                    />
                </div>
            );

        } else {
            return <div></div>
        }
    }

    render() {
        const scoringVMState = this.state.scoringGuideViewModel;

        if ((scoringVMState.kind == "success" || scoringVMState.kind == "reloading") && scoringVMState.content != undefined) {
            return (
                <div className="search-page">
                    <div className="search-container">
                        <ItemSearchContainer.ItemSearchContainer
                            onRowSelection={this.onRowSelection}
                            filterOptions={this.state.filterOptions}
                            searchClient={SearchClient}
                            item={this.state.item}
                        />
                    </div>
                </div>
            );
        }
        else {
            return <div></div>;
        }

    }
}


