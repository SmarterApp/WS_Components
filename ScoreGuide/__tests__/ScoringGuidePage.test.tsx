import * as React from 'react';
import * as $ from 'jquery'
import { ScoringGuidePage, ItemsSearchViewModel } from '../ScoringGuidePage';
import * as ItemModels from '../../Models/ItemModels';
import { shallow, mount } from 'enzyme';

const func = () => {
    const subjects: ItemModels.Subject[] = [];
    return new Promise<ItemsSearchViewModel>((resolve ,reject) => {
        let item: ItemsSearchViewModel = {subjects};
        resolve(item);
    })
}

describe('ScoringGuidePage', () => {

    it('matches snapshot', () => {
        expect(shallow(<ScoringGuidePage scoreGuideViewModelClient={func}/>)).toMatchSnapshot();
    });
});