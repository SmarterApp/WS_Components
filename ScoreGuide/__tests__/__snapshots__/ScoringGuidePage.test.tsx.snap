// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoringGuidePage matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div />,
  "nodes": Array [
    <div />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ScoringGuidePage
        scoreGuideViewModelClient={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ScoringGuidePage {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "getAboutItem": [Function],
        "loadScoringGuideViewModel": [Function],
        "onAboutItemError": [Function],
        "onAboutItemSuccess": [Function],
        "onErrorLoadScoringGuideViewModel": [Function],
        "onRowSelection": [Function],
        "onSuccessLoadScoringGuideViewModel": [Function],
        "props": Object {
          "scoreGuideViewModelClient": [Function],
        },
        "refs": Object {},
        "renderTabsContainer": [Function],
        "state": Object {
          "filterOptions": Object {
            "grades": Array [
              1,
              2,
              4,
              8,
              16,
              32,
              960,
            ],
            "subjects": Array [
              Object {
                "code": "ELA",
                "label": "English language arts / Literacy",
              },
              Object {
                "code": "MATH",
                "label": "Mathematics",
              },
            ],
            "techTypes": Array [
              Object {
                "code": "CAT",
                "label": "CAT",
              },
              Object {
                "code": "PT",
                "label": "Performance Items",
              },
            ],
          },
          "item": Object {
            "kind": "none",
          },
          "scoringGuideViewModel": Object {
            "kind": "loading",
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div />,
        "_debugID": 2,
        "_renderedOutput": <div />,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ScoringGuidePage
    scoreGuideViewModelClient={[Function]}
/>,
}
`;
